src/platforms/esp32/main/component_drivers.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+##
+## Copyright (c) 2020 fred@dushin.net
+## All rights reserved.
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+##
+
+import os
+import sys
+
+def load(path) :
+    with open(path) as f:
+        return f.read()
+
+def insert(component_driver, component_drivers_txt) :
+
```

Comment:
Maybe a comment here about the produced output might be a valuable addition.
Is it possible to use any terse templating syntax instead appending strings to buf?

Reply:
I have added a README.md explaining how the build works, for future reference.